src/macros.jl

```diff
@@ -2161,3 +2161,312 @@ julia> @by df :a begin
 macro by(x, what, args...)
     esc(by_helper(x, what, args...))
 end
+
+##############################################################################
+##
+## @unique - distinct row selection
+##
+##############################################################################
+
+function unique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    tmp = gensym("tmp")
+    quote            
+        $(tmp) = DataFrames.select($x, $(t...); copycols = false)
```

Comment:
Can you put this all into a separate function? That way we can dispatch on `::AbstractDataFrame` and avoid thinking about `GroupedDataFrame`s

Reply:
as in

```
function DataFramesMeta.make_unique(df, args...)
    ...
end

function unique_helper(x, args...)
    ...
    quote
        $make_unique($x, $t...)
    end
end
```